### PR TITLE
Trauma rework again

### PR DESCRIPTION
**Pull request by MaxIsJoe**
### Purpose
part of #6272 
Small rework by the request of @PerfectTangent . The changes are as follow : 
- Trauma damage is no longer tracked by numbers and is instead an effect that happens due to a fatal blow.
- Weapons that have the ability to inflict multiple traumas no longer apply multiple of them at once, they randomly get chosen from.
- Due to the removal of `cutSize`, Slash no longer contributes to the ability of dismbowling someone and is now fully reserved for pierce attacks.
- Blunt is now a tracked level rather than three bools. _(Still need to implement #7432 after this is merged)_
- Bodyparts get charred when in a serious condition instead of a critical one now.
- Pierce now causes external bleeding on the first level while slash only starts it on the second.
- Trauma applying armor chances only happen if your body is above light-moderate damage severity. This check is skipped for cases like a bomb explosion or extreme fire damage.
- Weapons no longer have the ability to have a 100% chance to cause trauma and the general balance philosophy when balancing weapons now is that weapons should only have a 0-50% chance for trauma rolls hitting. (2.5-10% for most common items, 15-25% for most weapons that make sense for them to cause trauma, 30%+ for special items, 50% for rare and items that specialize in dismemberment.) 

### Notes:
~~This PR requires extensive testing before merging and I'd like people to report any issues they find.~~
~~This PR should be good to go **after** all weapons and limbs have been rebalanced to the new current rework. Code reviews are welcome now.~~
**TIME TO MERGE** WOOHOO

### Changelog:
CL: [New] - Fully healing a body part will now cause Punctures to close on their own.
CL: [New] - Trauma is no longer tracked by numbers and are now caused by "fatal blows".
CL: [Balance] - Blood loss values has been reduced from 5-20 to 2-18
CL: [Balance] - Internal bleeding values has been reduced to 2-4 instead of 4-18
CL: [Balance] - Weapons can no longer apply multiple traumas at once.
CL: [Balance] - Weapons can no longer apply traumas unless the body part has received light moderate total damage.
CL: [Balance] - Bleeding is now caused by Minor Breakage and Open Lacerations.
CL: [Balance] - Increased default damage thresholds to 21 instead of 18
CL: [Balance] - You can no longer slash someone until their organs fall out.
CL: [Fix] - Burn damage visuals should now correctly display on Third Degree Burns.
CL: [Fix] - Fixed some issues where some traumas were not getting applied.
CL: [Fix] - Fixed an issue where Weeping Avulsions would cause bleeding to never stop.
CL: [Fix] - Fixed an issue where Prosthetics would have the ability to bleed.

### CODE REVIEW

**Comment by corp-0** [comment]
what makes a blow a fatal blow?

**Comment by MaxIsJoe** [comment]
> what makes a blow a fatal blow?

Chance and damage thresholds, though no numbers has changed yet so most weapons should be immediate for the time being until I've made sure that this thing actually works.

**Comment by MaxIsJoe** [comment]
Requesting code reviews as this PR seems to be nearly done.
Do not merge this until I've re-balanced all items and body parts to adapt to the new rework.

**Comment by MaxIsJoe** [comment]
Requesting reviews again, the last things are done and I've written the changelog.

**Review by NoooneyDude — DISMISSED** [review]
Some style guide violations.

**Comment by MaxIsJoe** [comment]
> Some style guide violations.

Where? I'm pretty sure I cleared some nesting and other stuff.

**Comment by NoooneyDude** [comment]
> > Some style guide violations.
> 
> Where? I'm pretty sure I cleared some nesting and other stuff.

`{`, `}` should be on their own lines.

What isn't mentioned is omitting them. At your discretion there, I suppose.

Personally I'll only omit them for jump statements.

**Comment by MaxIsJoe** [comment]
> > > Some style guide violations.
> > 
> > 
> > Where? I'm pretty sure I cleared some nesting and other stuff.
> 
> `{`, `}` should be on their own lines.
> 
> What isn't mentioned is omitting them. At your discretion there, I suppose.
> 
> Personally I'll only omit them for jump statements.

I think it's cleaner to have them specifically for things like `return` and very short things that don't require a whole new line for them.

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs [hunk 1]
```diff
@@ -255,18 +255,19 @@ public void AffectDamage(float damage, int damageType)
...
-				TakeBluntDamage(damage);
+				TakeBluntDamage();
 				if(damageToLimb >= DamageThreshold)
 				{
 					DismemberBodyPartWithChance();
 				}
 			}
 
-			if (attackType == AttackType.Fire || attackType == AttackType.Laser || attackType == AttackType.Energy)
+			if(Severity <= DamageSeverity.LightModerate){return;}
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]: I'm fine with returning in the same line, but in that case remove the brackets.

**Comment by corp-0** [comment]
I'm liking this new approach, should make things a lot easier to follow. If I'm completely honest, I never understood the trauma system in its entirety.

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]: Done 👍

**Review by NoooneyDude — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

**Pull request merged**